### PR TITLE
Return failing status code when TPU CI tests fail

### PR DESCRIPTION
**Pull request by will-cromar**
Currently, the TPU CI is not correctly returning a failing status when tests start failing. There has been a typo in the path to `test_operations.py` that we haven't noticed as a result.

Also, the test will fail if two pods schedule on the same TPU host (since only one can use the TPU device), so add an anti-affinity to prevent that from happening.

Tested by manually running this manifest with the test image from the last commit. The only failing test is this one:

```
======================================================================
FAIL: test_broadcast_master_param_synchronized_parameters (__main__.TestTpuCollectiveOps)
TestTpuCollectiveOps.test_broadcast_master_param_synchronized_parameters
test_broadcast_master_param_synchronized_parameters(True)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/absl/testing/parameterized.py", line 320, in bound_param_test
    return test_method(self, *testcase_params)
  File "pytorch/xla/test/pjrt/test_experimental_pjrt_tpu.py", line 255, in test_broadcast_master_param
    np.testing.assert_array_equal(master_params, worker_params)
  File "/usr/local/lib/python3.8/site-packages/numpy/testing/_private/utils.py", line 985, in assert_array_equal
    assert_array_compare(operator.__eq__, x, y, err_msg=err_msg,
  File "/usr/local/lib/python3.8/contextlib.py", line 75, in inner
    return func(*args, **kwds)
  File "/usr/local/lib/python3.8/site-packages/numpy/testing/_private/utils.py", line 862, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Arrays are not equal

Mismatched elements: 25 / 25 (100%)
Max absolute difference: 0.42102337
Max relative difference: inf
 x: array([[ 0.230433, -0.19739 , -0.086697,  0.209908, -0.421023],
       [ 0.268202, -0.092003,  0.227517,  0.06217 , -0.054759],
       [ 0.12404 ,  0.022062,  0.163335, -0.17428 , -0.032606],...
 y: array([[ 0.,  0.,  0., -0.,  0.],
       [ 0.,  0.,  0., -0., -0.],
       [-0.,  0., -0.,  0., -0.],...

----------------------------------------------------------------------
```

This test started failing between 9e4b01af28bb607ac79c2ce4f35d52623c854011 and f594c29ac94a979d4541c44d334a2870417c628e, but the breakage is obscured by a build failure in that range. I'll investigate and follow up with a fix.

Tested the anti-affinity rule by deploying two copies of the workload and confirming the second has ` Warning  FailedScheduling   [...] 1 node(s) didn't match pod anti-affinity rules [...]`

### CODE REVIEW

File: test/tpu/xla_test_job.yaml [hunk 1]
```diff
@@ -25,9 +38,9 @@ spec:
     image: gcr.io/$PROJECT_ID/pytorch-xla-test:$BUILD_ID
     command:
     - bash
-    - -c
+    - -cxe
     - |
-      python3 pytorch/xla/test/pjrt/test_operations.py -v
```
Comment (vanbasten23) [inline test/tpu/xla_test_job.yaml:30]: before the change, wouldn't it throw a file not found error?

**Review by JackCaoG — APPROVED** [review]

File: test/tpu/xla_test_job.yaml (hunk 1 above)
Comment (will-cromar) [reply test/tpu/xla_test_job.yaml:30]: Right, but the failing status was ignored. That's fixed by setting the `-e` flag in the bash script

**Comment by will-cromar** [comment]
GPU failure is unrelated

File: test/tpu/xla_test_job.yaml (hunk 1 above)
Comment (vanbasten23) [reply test/tpu/xla_test_job.yaml:30]: thanks!

**Review by vanbasten23 — APPROVED** [review]

**Comment by will-cromar** [comment]
Actually, there are still failures in the CI that are getting ignored here

```
======================================================================
ERROR: test_broadcast_master_param_synchronized_parameters (__main__.TestTpuCollectiveOps)
TestTpuCollectiveOps.test_broadcast_master_param_synchronized_parameters
test_broadcast_master_param_synchronized_parameters(True)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/absl/testing/parameterized.py", line 320, in bound_param_test
    return test_method(self, *testcase_params)
  File "pytorch/xla/test/pjrt/test_experimental_pjrt_tpu.py", line 252, in test_broadcast_master_param
    self.skipIf(sync)
AttributeError: 'TestTpuCollectiveOps' object has no attribute 'skipIf'

======================================================================
ERROR: test_broadcast_master_param_unsynchronized_parameters (__main__.TestTpuCollectiveOps)
TestTpuCollectiveOps.test_broadcast_master_param_unsynchronized_parameters
test_broadcast_master_param_unsynchronized_parameters(False)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/absl/testing/parameterized.py", line 320, in bound_param_test
    return test_method(self, *testcase_params)
  File "pytorch/xla/test/pjrt/test_experimental_pjrt_tpu.py", line 252, in test_broadcast_master_param
    self.skipIf(sync)
AttributeError: 'TestTpuCollectiveOps' object has no attribute 'skipIf'

----------------------------------------------------------------------
```

Taking this back into draft

**Comment by will-cromar** [comment]
See 718e401 for example failing status :confetti_ball:

**Pull request merged**